src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -738,10 +738,16 @@ void MessageHelper::SendDeleteChoiceSetRequest(smart_objects::SmartObject* cmd,
   using namespace smart_objects;
 
   SmartObject msg_params = SmartObject(smart_objects::SmartType_Map);
-
   msg_params[strings::app_id] = application->app_id();
   msg_params[strings::type] = hmi_apis::Common_VRCommandType::Choice;
+
+  if (!cmd->keyExists(strings::grammar_id) ||
+      (*cmd)[strings::grammar_id].asInt() == -1) {
+    return;
+  }
```

Comment:
Same here, probably wouldn't hurt to put this check toward the top of the function before we start constructing things